benchto-driver/src/main/java/com/teradata/benchto/driver/loader/BenchmarkLoader.java

```diff
@@ -139,32 +137,65 @@
         }
     }
 
-    private List<Path> findBenchmarkFiles()
+    private List<Resource> findBenchmarkFiles(ResourcePatternResolver patternResolver)
             throws IOException
     {
-        LOGGER.info("Searching for benchmarks in classpath ...");
+        String benchmarksDir = properties.getBenchmarksDir();
+        List<String> activeBenchmarkPatterns = activeBenchmarkPatterns();
 
-        List<Path> benchmarkFiles = Files
-                .walk(properties.benchmarksFilesPath())
-                .filter(file -> isRegularFile(file) && file.toString().endsWith(BENCHMARK_FILE_SUFFIX))
+        LOGGER.info("Searching for {} benchmarks in  {} ...", activeBenchmarkPatterns, benchmarksDir);
+
+        List<Resource> benchmarkFiles = activeBenchmarkPatterns.stream()
+                .flatMap(pattern -> getResources(patternResolver, benchmarksDir + "/" + pattern))
+                .peek(resource -> LOGGER.info("Benchmark found: {}", resource))
                 .collect(toList());
-        benchmarkFiles.stream().forEach((path) -> LOGGER.info("Benchmark found: {}", path.toString()));
 
         return benchmarkFiles;
     }
 
-    private List<Benchmark> loadBenchmarks(String sequenceId, List<Path> benchmarkFiles)
+    private ResourcePatternResolver getResourcePatternResolver()
+    {
+        FileSystemResourceLoader resourceLoader = new FileSystemResourceLoader()
+        {
+            @Override
+            protected Resource getResourceByPath(String path)
+            {
+                if (path != null && path.startsWith("/")) {
+                    // Workaround superclass's stripping of leading '/'. Shameful.
+                    path = "/" + path;
+                }
+                return super.getResourceByPath(path);
+            }
+        };
+
+        resourceLoader.setClassLoader(Thread.currentThread().getContextClassLoader());
+        return new PathMatchingResourcePatternResolver(resourceLoader);
+    }
+
+    private static Stream<Resource> getResources(ResourcePatternResolver patternResolver, String locationPattern)
```

Comment:
Generally I think you could wrap usage of `ResourcePR` as separate class where you could expose only methods which should be used from here. Then you new class API could not to have any `throws` clause. 

IMO it would be nice to do this that way as this class would be less polluted with different problems. Would be more focuses on its main responsibility and not tools it is using.